### PR TITLE
Refactor BoltInstanceTest and SpoutInstanceTest to not use Thread.sleep()

### PR DESCRIPTION
**Pull request by billonahill**
- Removes the `TestCommunicator` class in favor or `CommunicatorTestHelper.CommunicatorTestHelper` which is a better approach.
- Added more latches to `TestCommunicator`
- Updated Bolt/Spout instance tests to use countdown latches instead of checking static variables and sleeping

### CODE REVIEW

File: heron/instance/tests/java/com/twitter/heron/instance/bolt/BoltInstanceTest.java [hunk 1]
```diff
@@ -92,9 +93,12 @@ public void testReadTupleAndExecute() {
...
     slaveTester.getInControlQueue().offer(instanceControlMsg);
 
+    int expectedTuples = 10;
+    CountDownLatch executeLatch = new CountDownLatch(expectedTuples);
     SingletonRegistry.INSTANCE.registerSingleton(Constants.ACK_COUNT, ackCount);
     SingletonRegistry.INSTANCE.registerSingleton(Constants.FAIL_COUNT, failCount);
-    SingletonRegistry.INSTANCE.registerSingleton("execute-count", tupleExecutedCount);
+    SingletonRegistry.INSTANCE.registerSingleton(Constants.EXECUTE_COUNT, tupleExecutedCount);
+    SingletonRegistry.INSTANCE.registerSingleton(Constants.EXECUTE_LATCH, executeLatch);
     SingletonRegistry.INSTANCE.registerSingleton("received-string-list", receivedStrings);
```
Comment (nlu90) [inline heron/instance/tests/java/com/twitter/heron/instance/bolt/BoltInstanceTest.java]: also move this to `Constant`?

Comment (billonahill) [inline heron/instance/tests/java/com/twitter/heron/instance/bolt/BoltInstanceTest.java]: Cool catch, updated.

File: heron/instance/tests/java/com/twitter/heron/instance/bolt/BoltInstanceTest.java [hunk 2]
```diff
@@ -92,10 +93,13 @@ public void testReadTupleAndExecute() {
 
     slaveTester.getInControlQueue().offer(instanceControlMsg);
 
+    int expectedTuples = 10;
```
Comment (objmagic) [inline heron/instance/tests/java/com/twitter/heron/instance/bolt/BoltInstanceTest.java]: `final int`?

**Comment by billonahill** [comment]
Good call @objmagic, updated.

**Pull request merged**